CRM/Contribute/BAO/Contribution.php

```diff
@@ -3469,22 +3469,10 @@ public static function updateFinancialAccounts(&$params, $context = NULL, $skipT
         $deferredFinancialAccount = CRM_Contribute_PseudoConstant::financialAccountType($params['prevContribution']->financial_type_id, $relationTypeId);
       }
       $lastFinancialTrxnId = CRM_Core_BAO_FinancialTrxn::getFinancialTrxnId($params['prevContribution']->id, 'DESC', FALSE, NULL, $deferredFinancialAccount);
-      if ($params['trxnParams']['total_amount'] < 0) {
-        if (!empty($lastFinancialTrxnId['financialTrxnId'])) {
-          if ($params['total_amount'] > 0) {
-            $params['trxnParams']['to_financial_account_id'] = CRM_Core_DAO::getFieldValue('CRM_Financial_DAO_FinancialTrxn', $lastFinancialTrxnId['financialTrxnId'], 'to_financial_account_id');
-            $params['trxnParams']['payment_instrument_id'] = $params['prevContribution']->payment_instrument_id;
-          }
-          else {
-            $params['trxnParams']['to_financial_account_id'] = $params['to_financial_account_id'];
-            $params['trxnParams']['payment_instrument_id'] = $params['contribution']->payment_instrument_id;
-          }
-        }
-      }
-      else {
-        if ($params['total_amount'] < 0) {
-          $params['trxnParams']['payment_instrument_id'] = $params['prevContribution']->payment_instrument_id;
+      if (!empty($lastFinancialTrxnId['financialTrxnId'])) {
+        if ($params['total_amount'] != $params['trxnParams']['total_amount']) {
           $params['trxnParams']['to_financial_account_id'] = CRM_Core_DAO::getFieldValue('CRM_Financial_DAO_FinancialTrxn', $lastFinancialTrxnId['financialTrxnId'], 'to_financial_account_id');
+          $params['trxnParams']['payment_instrument_id'] = $params['prevContribution']->payment_instrument_id;
         }
         else {
```

Comment:
@pradpnayak the only question i have is if this should be the else for the !empty rather than the 2nd if i have done some tests and this works but just not sure if we are missing something